mex/editor/rules/local_storage_mixin_state.py

```diff
@@ -0,0 +1,88 @@
+import reflex as rx
+
+from mex.editor.rules.models import (
+    LocalDraft,
+    LocalDraftStorageObject,
+    LocalEdit,
+    LocalEditStorageObject,
+    UserDraft,
+    UserDraftSummary,
+    UserEdit,
+)
+from mex.editor.transform import transform_fields_to_title
+
+
+class LocalStorageMixinState(rx.State, mixin=True):
+    """State-Mixin for handling local drafts and edits of EditorFields."""
+
+    local_draft_store: str = rx.LocalStorage('{"value": {}}', sync=True)
+    local_edit_store: str = rx.LocalStorage('{"value": {}}', sync=True)
```

Comment:
why does this need to be a mixin when it is used only once in the RuleState? neither the RuleState nor this mixin would work on its own: the RuleState has a dependency to the Mixin because `update_local_state` wants to call `self.update_draft()`. and the Mixin has a dependency on the business-logic level to the RuleState, because it can only handle UserDraft and UserEdit models. (so it couldn't easily be used by the ingest page for example.)